internal/pkg/source/source.go

```diff
@@ -109,13 +109,25 @@ func (a *Source) visitOperands(operands []*ssa.Value) {
 		if !ok || a.marked[n] {
 			continue
 		}
-		al, ok := (*o).(*ssa.Alloc)
-		if !ok || al.Comment == "slicelit" || a.config.IsSource(al.Type()) {
-			a.dfs(n)
+		al, isAlloc := (*o).(*ssa.Alloc)
+		// An Alloc represents the allocation of space for a variable. If a Node is an Alloc,
+		// and the thing being allocated is not a slice, then either:
+		// a) it is a Source value, in which case it will get its own DFS
+		// b) it is not a Source value, in which case we should not DFS through it.
+		// However, if the Alloc is a slice, then that means the source that we are DFSing from
+		// is being placed into that slice, so we do need to keep DFSing.
+		if isAlloc && !isSlice(utils.Dereference(al.Type())) {
+			return
 		}
+		a.dfs(n)
 	}
 }
 
+func isSlice(t types.Type) bool {
+	_, ok := t.(*types.Array)
```

Comment:
Is slice checks if the type is an Array?  Not `types.Slice`?

Reply:
I think the ssa allocates a `types.Array` as the backing array for a slice. Since I've inlined the code `isSlice` became a variable. In retrospect in think `isArray` is less confusing. I've also updated the doc comment.

(The reason I called it `isSlice` initially is that in actual code usually the `Array` is intended to be used for a slice, but that is a poor justification.)

Reply:
> Since I've inlined the code

That makes this dead, then.